core/src/trezor/strings.py

```diff
@@ -64,3 +70,57 @@ def format_duration_ms(milliseconds: int) -> str:
         divisor = 1
 
     return format_plural("{count} {plural}", milliseconds // divisor, unit)
+
+
+def _calculate_timestamp_correction(timestamp: int) -> (Tuple[tuple, int]):
+    """
+    utime module can't convert timestamp to datetime with seconds precision.
```

Comment:
It looks like this problem occurs only in the UNIX implementation, because micropython quite pointlessly converts the timestamp to a C `float` which has a mantissa of 23 bits, causing the lower bits to get lost. The STM32 implementation reads it as an integer.

Reply:
This is actually fixed upstream.

Reply:
Yes, we need to merge MicroPython 1.14 in before merging in this PR.